moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/action_based_controller_handle.h

```diff
@@ -136,11 +138,29 @@ class ActionBasedControllerHandle : public ActionBasedControllerHandleBase
     }
     else
     {
-      std::future_status status = result_future.wait_for(timeout.to_chrono<std::chrono::duration<double>>());
-      if (status == std::future_status::timeout)
+      const bool use_sim_time = node_->get_parameter("use_sim_time").as_bool();
+      if (use_sim_time)
+      {
+        std::future_status status;
```

Comment:
Since the if() and the else() both use `status`, you could define it at Line 141 (before the if/else).

(I'll try to push this trivial nitpick to your branch)